### PR TITLE
Refine panel and step 2

### LINKED ISSUE #84: Refine details panel and step2
- [ ] Add start and end date for the campaign in details panel
- [x] Hide Short name and code in details panel
- [x] Add description to step 2

### PR DESCRIPTION
**Pull request by adrianq**
### :pushpin: References

* **Issue:** Closes #84

### :memo: Implementation

- Remove code and shortname and add start and end date in details panel
- Add description to step 2

### :art: Screenshots
image
image

### :fire: Is there anything the reviewer should know to test it?
I have some doubts about how to implement the start and end date. Hope the implementation is not too dirty...

### :bookmark_tabs: Others

- Any change in the [metadata repo](https://github.com/EyeSeeTea/vaccination-metadata)? If so, what branch/PR?
- Any change in the [GUI library](https://github.com/EyeSeeTea/d2-ui-components)? If so, what branch/PR?

### CODE REVIEW

**Comment by adrianq** [comment]
Ready for review!

File: src/models/datasets.js [hunk 1]
```diff
@@ -37,6 +37,13 @@ export async function list(config, d2, filters, pagination) {
     ]);
     const listOptions = { fields, filter, page, pageSize, order };
     const collection = await d2.models.dataSets.list(cleanOptions(listOptions));
+    collection.toArray().forEach(dataSet => {
+        const dataInputPeriods = dataSet.dataInputPeriods;
+        if (!_(dataInputPeriods).isEmpty()) {
+            dataSet.startDate = formatDateLong(dataInputPeriods[0].openingDate);
+            dataSet.endDate = formatDateLong(dataInputPeriods[0].closingDate);
+        }
+    });
```
Comment (tokland) [inline src/models/datasets.js]: Sorry, I should've pointed in the ticket to the (optional) attribute `getValue` of `detailFields`. This way we avoid adding properties in-place to model objects, which can be confusing when using them, and instead pass the details component a getter.

https://github.com/EyeSeeTea/d2-ui-components/blob/master/src/details-box/DetailsBox.component.js#L15

File: src/components/steps/general-info/GeneralInfoStep.jsx [hunk 2]
```diff
@@ -65,6 +68,17 @@ class GeneralInfoStep extends React.Component {
...
+            {
+                name: "description",
+                value: campaign.description,
+                component: TextField,
+                props: {
+                    floatingLabelText: i18n.t("Description"),
+                    style: { width: "33%" },
+                    changeEvent: "onBlur",
+                    "data-field": "description",
+                },
```
Comment (tokland) [inline src/components/steps/general-info/GeneralInfoStep.jsx]: Proposal: Add `multiLine: true`

File: src/models/datasets.js (hunk 1 above)
Comment (adrianq) [reply src/models/datasets.js]: No worries, I thought about asking but didn't want to bother you during the weekend :/ Should be ready now

**Pull request merged**